specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+        {
+            // If we successfully registered this instance, we can now just
+            // go ahead and do normal initialization.
+            RegisterClassAndStartMessagePump(hInstance, nCmdShow);
+        }
+        else
+        {
+            // Some other instance has already registered for this file,
+            // so we'll redirect this activation to that instance instead.
+            instance.RedirectTo(activationArgs);
+        }
+    }
+    return 1;
+}
+```
+
+### Arbitrary redirection
+
+In the previous example, the app only redirects for file activations. In the simplest case, it might
+just redirect on every activation if there is already an instance running. The logic for deciding
+when to redirect is entirely up to the app - it might be dependent on activation kind, or on some
+other logic.
+
+In the following example, the app might have checked some specific activation kind first, or it
+might not. Regardless, the app then checks all other running instances to see if any of them has
+registered to be a "reusable" instance. What constitutes "reusable" is also entirely up to the app
+to decide. It might be any instance that doesn't have a file open, or doesn't have an editing
+session active, or has some specific database connection, and so on. If it finds a reusable
+instance, it redirects activation to that instance; if not, it goes ahead with this activation and
+normal app initialization. In this example, the app signals that an instance is reusable by
+specifying the "reusable" text as the key it registers with the system. Of course, exactly what text
+to use for this purpose is entirely up to the app to decide. For more complex scenarios, if the app
+wanted to allow for multiple reusable instances, it could specify "reusable" plus some unique
+identifier for each instance, and so on.
+
+```c++
+int APIENTRY B_wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    ActivationArguments activationArgs =
+        AppInstance::GetCurrent().GetActivatedEventArgs();
+
+    // As above, check for any specific activation kind we care about.
+    ExtendedActivationKind kind = activationArgs.Kind;
+    if (kind == ExtendedActivationKind::File)
+    {
+        // etc... as in previous scenario.
+    }
+    else
+    {
+        // For other activation kinds, we'll trawl all instances to see if
+        // any are suitable for redirecting this request. First, get a list
+        // of all running instances of this app.
+        auto instances = AppInstance::GetInstances();
+
+        // In the simple case, we'll redirect to any other instance.
+        AppInstance instance = instances.GetAt(0);
+
+        // If the app re-registers re-usable instances, we can filter for these instead.
+        bool isFound = false;
+        for (AppInstance instance : instances)
+        {
+            if (instance.Key.c_str() == "REUSABLE")
```

Comment:
Code as written doesn't compile and even after adding necessary syntax, it would always be false because the `c_str()` will never match the embedded literal.

```suggestion
            if (instance.Key() == L"REUSABLE")
```

Reply:
SAMPLE: Fix

Reply:
Fixed.